.travis.yml

```diff
@@ -32,31 +35,19 @@ before_install:
 
 install:
  - travis_retry cabal update
- # can't use "cabal install --only-dependencies --enable-tests" due to dep-cycle
- - cabal install
-      "QuickCheck >=2.10 && <2.14"
-      "byteorder ==1.0.*"
-      "dlist >=0.5 && <0.9"
-      "mtl >=2.0 && <2.3"
-      deepseq test-framework-hunit test-framework-quickcheck2
 
 script:
- - cabal configure --enable-tests -v2
- - cabal build
- # --show-details=streaming is available for CABALVER>=1.20 only
- - if [ "$(echo -e "1.20\n$CABALVER" | sort -rV | head -n1)" = "$CABALVER" ]; then
-     cabal test --show-details=streaming;
-   else
-     travis_wait cabal test --show-details=always;
-   fi
- - cabal sdist
+ - cabal v2-configure; cabal v2-build
+ - cabal v2-sdist -o .
+ # Remove to enable build of sdist.
+ - rm -f cabal.project
+ - (cd tests; cabal v2-configure; cabal v2-test)
  # "cabal check" disabled due to -O2 warning
- - export SRC_TGZ=$(cabal info . | awk '{print $2 ".tar.gz";exit}') ;
-   cd dist/;
-   if [ -f "$SRC_TGZ" ]; then
-      cabal install --force-reinstalls "$SRC_TGZ";
+ - export SRC=$(cabal info . | awk '{print $2; exit}');
+   if [ -f "$SRC.tar.gz" ]; then
+      cabal get "./$SRC.tar.gz";
+      (cd "$SRC"; cabal v2-build);
    else
       echo "expected '$SRC_TGZ' not found";
```

Comment:
Oops, the error message here is not quite right, it should be `$SRC.tar.gz` not `$SRC_TGZ` which is no longer set.  This should not happen, but if it ever did, the error message would be confusing...

Reply:
@sjakobi Would you care to fix the error message?

Reply:
I'm not too worried about it. Feel free to send a PR or open an issue.